core/whistle_number_manager-1.0.0/src/wh_port_request.erl

```diff
@@ -200,3 +203,126 @@ enable_number(N) ->
+        {'error', _R} ->
+            lager:error("failed to open view port_requests/listing_submitted ~p", [_R]);
+        {'ok', JObjs} ->
+            Docs = [wh_json:get_value(<<"doc">>, JObj) || JObj <- JObjs],
+            maybe_send_requests(Docs)
+    end.
+
+-spec maybe_send_requests(wh_json:objects()) -> 'ok'.
+-spec maybe_send_requests(ne_binary() | 'undefined', wh_json:object()) -> 'ok'.
+maybe_send_requests([]) -> 'ok';
+maybe_send_requests([JObj|JObjs]) ->
+    Id = wh_json:get_value(<<"_id">>, JObj),
+    erlang:put('callid', Id),
+
+    AccountId = wh_json:get_value(<<"pvt_account_id">>, JObj),
+    AccountDb = wh_util:format_account_id(AccountId, 'encoded'),
+    case couch_mgr:open_doc(AccountDb, AccountId) of
+        {'error', _R} ->
+            lager:error("failed to open account ~s:~p", [AccountId, _R]);
+        {'ok', Doc} ->
+            Url = wh_json:get_value(<<"submitted_port_requests_url">>, Doc),
+            maybe_send_requests(Url, JObj)
+    end,
+    maybe_send_requests(JObjs).
+
+maybe_send_requests('undefined', JObj)->
+    AccountId = wh_json:get_value(<<"pvt_account_id">>, JObj),
+    lager:debug("'submitted_port_requests_url' is not set for account ~s", [AccountId]);
+maybe_send_requests(Url, JObj)->
+    case send_request(Url, JObj) of
+        'error' -> 'ok';
+        'ok' ->
+            case send_attachements(Url, JObj) of
+                'error' -> 'ok';
+                'ok' -> set_flag(JObj)
+            end
+    end.
+
+-spec send_request(ne_binary(), wh_json:object()) -> 'error' | 'ok'.
+send_request(Url, JObj) ->
+    Id = wh_json:get_value(<<"_id">>, JObj),
+
+    Headers = [{"Content-Type", "application/json"}
+               ,{"User-Agent", wh_util:to_list(erlang:node())}
+              ],
+
+    Uri = binary:bin_to_list(<<Url/binary, "/", Id/binary>>),
+
+    Remove = [
+        <<"_rev">>
+        ,<<"ui_metadata">>
+        ,<<"_attachments">>
+        ,<<"pvt_request_id">>
+        ,<<"pvt_type">>
+        ,<<"pvt_vsn">>
+        ,<<"pvt_account_db">>
+    ],
+    Replace = [
+        {<<"_id">>, <<"id">>}
+        ,{<<"pvt_port_state">>, <<"port_state">>}
+        ,{<<"pvt_account_id">>, <<"account_id">>}
+        ,{<<"pvt_created">>, <<"created">>}
+        ,{<<"pvt_modified">>, <<"modified">>}
+    ],
+    Data = wh_json:encode(wh_json:normalize_jobj(JObj, Remove, Replace)),
+
+    case ibrowse:send_req(Uri, Headers, 'post', Data, []) of
+        {'ok', "200", _Headers, _Resp} ->
+            lager:debug("submitted_port_request successfully sent");
+        _Other ->
+            lager:error("failed to send submitted_port_request ~p", [_Other]),
+            'error'
+    end.
+
+-spec send_attachements(ne_binary(), wh_json:object()) -> 'error' | 'ok'.
+send_attachements(Url, JObj) ->
+    Id = wh_json:get_value(<<"_id">>, JObj),
+    Attachments = wh_json:get_value(<<"_attachments">>, JObj),
+    wh_json:foldl(
```

Comment:
This fold will only return the result of the last attachment upload attempt. If there are 3 attachments, and the first one fails, should the rest be attempted anyway? If so, and they succeed, should 'ok' really be returned?